apps/labs/posts/polars-plugins-lets-make-them-easier-to-use.md

```diff
@@ -0,0 +1,161 @@
+---
+title: "Polars Plugins: let's make them easier to use"
+description: 'Increasing ease-of-use of Polars plugins by improving an existing tutorial.'
+published: September 30, 2024
+authors: [bruno-kind]
+category: [OSS Experience, Internship]
+featuredImage:
+  src: /posts/polars-plugins-lets-make-them-easier-to-use/featured.png
+  alt: 'Polar bear with a yellow safety helmet and a red hammer on the ground, representing work-in-progress.'
+hero:
+  imageSrc: /posts/polars-plugins-lets-make-them-easier-to-use/hero.png
+  imageAlt: 'Polar bear with a yellow safety helmet and a red hammer on the ground, with the letters W.I.P., for work-in-progress.'
+---
+
+I am Bruno Kind, and this is the place where I tell the world about my journey in Open Source during July-September 2024.
+How I got an internship to work with Open Source, in an entry-level Rust position – yes, they're real!
+And how I managed to contribute and improve Marco Gorelli's existing [Polars Plugin Tutorial](https://marcogorelli.github.io/polars-plugins-tutorial/).
+Hop in, grab your popcorn, fasten your seatbelts.
+
+---
+
+Sometime before the internship began, I was told my project would be related to Polars plugins.
+Thrilled, I rushed and wrote my first plugin, following the official tutorial, then... nothing.
+I'm not a Polars user, nor is my Rust-fu strong enough to explore the plugin system in-depth.
+But I did have an idea - what if I could make Conway's Game of Life play itself in a dataframe?
+Could plugins do that? Regardless, I had to wait. I lacked the juice.
+
+Time passed, as it always does, and the internship began.
+I had my first days, met **Marco** (my mentor), talked to people from different areas, followed all the steps from the onboarding process, yada yada.
+It didn't take long for my end-goal to be presented to me, which was: **to improve and make it easier to follow a Polars Plugin tutorial** (the one authored by Marco).
+
+I was having daily meetings with him, and he was encouraging me to follow the tutorial as well as explore plugins in general.
+And so I did. I went through the entire tutorial in the first days of the week, and eventually decided to share with him the crazy idea I had before it all started - making Conway's Game of Life "run" in a dataframe.
+What would you tell a lunatic intern with such a ridiculous idea? I don't know, but he said something along the lines of "I'm totally on board with it".
+This was **it**, _the game was on!_ But wait, no... I don't know how to do that... I actually have no clue!
+
+Those were my thoughts, initially. I believed the same limitations I had before the internship began were also present here, but that was not entirely true.
+After all, I did the tutorial, I played with some examples, and I even managed to "explain" ChatGPT a broader context of plugins, allowing it to give me some useful suggestions besides its usual hallucinations.
```

Comment:
This sentence seems a bit off. I think you meant
```suggestion
After all, I did the tutorial, I played with some examples, and I even managed to "explain" to ChatGPT a broader context of plugins, allowing it to give me some useful suggestions besides its usual hallucinations.
```

Reply:
You're right, thanks!